src/main/java/com/celements/web/plugin/cmd/DocFormCommand.java

```diff
@@ -116,6 +118,52 @@
     }
     return new HashSet<XWikiDocument>(changedDocs.values());
   }
+  
+  public Map<String, String[]> prepareMapForDocUpdate(Map<String, ?> map) {
+    Map<String, String[]> recompMap = new HashMap<String, String[]>();
+    for (String key : map.keySet()) {
+      if(map.get(key) instanceof String[]) {
+        recompMap.put(key, (String[])map.get(key));
+      } else if(map.get(key) instanceof String) {
+        recompMap.put(key, new String[]{(String)map.get(key)});
+      }
+    }
+    if (LOGGER.isTraceEnabled()) {
+      LOGGER.trace("recompiled map '{}' to '{}'", map, recompMap);
+    }
+    return recompMap;
+  }
+
+  public Map<String, Set<DocumentReference>> saveXWikiDocCollection(
+      Collection<XWikiDocument> xdocs) {
+    Set<DocumentReference> savedSuccessfully = new HashSet<DocumentReference>();
+    Set<DocumentReference> saveFailed = new HashSet<DocumentReference>();
+    for (XWikiDocument xdoc : xdocs) {
+      if(!xdoc.isNew() || "true".equals(getContext().getRequest().get("createIfNotExists"))) {
+        try {
+          if(getContext().getWiki().getRightService().hasAccessLevel("edit", 
+              getContext().getUser(), getWebUtilsService().serializeRef(
```

Comment:
Please move rights checks to script service or api methods. Components may not additionally check for rights.